### ISSUE TITLE
A general question about PGP/MIME.

### ISSUE THREAD

**Issue by dirkmittler** [kylehuff/webpg-chrome#106]
The main new feature which v0.9.4 seems to introduce (over v0.9.3) is PGP/MIME support. And the user activates this, by Linking his GMail Identity, which causes a separate browser window to appear (at least it did in my case), with a Google URL, in which the user is asked whether he'd like the app to have raw access to his email. This counts as an additional log-in for that user, as if the app was one of his devices. Attempts can also be made by Google, to geo-locate that...

Now, the Dev has enlightened us, by telling us that in the object-oriented code he works with, this creates an ' XOAUTH2 ' object.

My question is this: Does WebPG verify online, on a per-use basis, whether the sign-in key this object holds is still valid? It seems to me that the extension should test this additional log-in _at least every time the user opens the Properties panel_ , because any email could have a(n otherwise invisible) PGP/MIME attachment, and because it's an object which can be deleted externally, without notification, by Google, the same way they can just randomly force any device to resubmit log-in credentials.

And from what I'm gathering, if this key is no longer valid, then also, any advanced effort WebPG v0.9.4 makes to encrypt, or to sign, will not work when the user has selected it from the WebPG button, ?

Also, it would be unrealistic to expect the user to renew this manually in the WebPG Properties panel (where his Google-selfie could just be a locally-cached image), because the user is never notified, that one of his log-ins has been cancelled. When Google cancels such a log-in, it's their assumption that a valid user will notice this, the next time he tries to use it. But because WebPG generates no error messages, average users _won't_ notice this...

Dirk

[Edit] You see, I've had no problem getting FF v0.9.3 to work, both under Windows and under Linux with Iceweasel. The question strikes me as valid, of why versions 0.9.3 would work with no reliability issues in any form, while versions 0.9.4 would be having such a hard time, in my case, which other people have a hard time to reproduce. With FF v0.9.3 , even the WebPG button appears correctly to the left of the Send button, which under v0.9.4 it did not.

I don't really assume either, that the Dev has made grotesque coding errors between these two versions, because by now somebody would have seen them. And so to my mind, the answer could be something subtle, like what I just wrote.

![webpg dual send buttons _7](https://cloud.githubusercontent.com/assets/7936371/3349934/5193f766-f983-11e3-93b7-f752e8fc1f0b.jpg)

Now, the way I last learned object-oriented programming, an attempt to initialize some primary object, can invoke an innocent attempt to initialize some secondary object. But then if the attempt for the secondary object fails, this can throw an exception, which can cause the initialization of the primary object also to fail...

In short, your code may never have attempted to draw two Send buttons. It may simply have failed to initialize some send functionality, in a way that prevented the WebPG button from being drawn (in v0.9.4 , as I complained about in thread #105 . )
